tools/integration_tests/setup/setup.go

```diff
@@ -183,32 +215,73 @@ func ExecuteTest(m *testing.M) (successCode int) {
 	return successCode
 }
 
+func executeTestsForMountedDirectory(flags []string, m *testing.M) (successCode int) {
+	// Clean the mountedDirectory before running any tests.
+	os.RemoveAll(mntDir)
+
+	successCode = ExecuteTest(m)
+
+	err := UnMount()
+	if err != nil {
+		LogAndExit(fmt.Sprintf("Error in unmounting bucket: %v", err))
+	}
+
+	// Print flag on which test fails
+	if successCode != 0 {
+		f := strings.Join(flags, " ")
+		log.Print("Test Fails on " + f)
+		return
+	}
+	return
+}
+
 func ExecuteTestForFlags(flags [][]string, m *testing.M) (successCode int) {
 	var err error
 
+	// Run tests for static mounting.
 	for i := 0; i < len(flags); i++ {
-		if err = MountGcsfuse(flags[i]); err != nil {
+		if err = MountGcsfuseWithStaticMounting(flags[i]); err != nil {
 			LogAndExit(fmt.Sprintf("mountGcsfuse: %v\n", err))
 		}
+		successCode = executeTestsForMountedDirectory(flags[i], m)
+	}
 
-		// Clean the mountedDirectory before running any tests.
-		os.RemoveAll(mntDir)
+	// Run tests for mounting a specific directory in a Cloud Storage bucket instead of the entire bucket, where directory not exist in the bucket.
+	for i := 0; i < len(flags); i++ {
+		if err = MountGcsfuseWithOnlyDir(flags[i], DirectoryInTestBucket); err != nil {
```

Comment:
I think this is getting messy(say in future we want to run some but not all test cases for more similar flags then it would be difficult to add them like this). Ideally, we should have a file/dir for --only-dir flag and whichever existing tests we want to run for that flag should be invoked from there.